src/Common/AzurePSCmdlet.cs

```diff
@@ -254,7 +257,7 @@ protected bool IsVerbose()
 
         protected new void WriteError(ErrorRecord errorRecord)
         {
-            FlushDebugMessages();
+            FlushDebugMessages(IsDataCollectionAllowed());
```

Comment:
But we only want to log when there is an exception. Do you mean you want to log everything if data collection is allowed?